### PR TITLE
Example for References

### PR DESCRIPTION
**Pull request by masseelch**

### CODE REVIEW

File: dsl_test.go [hunk 1]
```diff
@@ -29,14 +53,55 @@ func TestBuilder(t *testing.T) {
...
+			},
+		},
+		Components: &ogen.Components{
+			Schemas:   nil, // TODO
+			Responses: nil, // TODO
+			Parameters: map[string]ogen.Parameter{
+				authQ.Ident(): *authQ.Parameter,
+				authH.Ident(): *authH.Parameter,
+				csrf.Ident():  *csrf.Parameter,
+			},
```
Comment (a8m) [inline dsl_test.go]: Maybe provide a cleaner API for declaring them?

```go
Parameters: ogen.NamedParamaters(authQ, authH, csrf),
```

File: dsl_test.go [hunk 2]
```diff
@@ -7,7 +7,31 @@ import (
...
+		"authInHeader",
+		ogen.NewParameter().
+			SetIn("header").
+			SetName("Authorization").
+			SetDescription("Optional bearer token"),
+	)
+	// referenced cookie param
+	csrf := ogen.NewParameter().InCookie().
+		SetName("csrf").
+		SetDescription("CSRF token").ToNamedParameter("csrf")
```
Comment (a8m) [inline dsl_test.go]: Not sure I have the full context, but did you consider using a less-verbose method like:

```go
csrf := ogen.NewParameter().SetIn(...).SetDescription(...).As("csrf")
```

File: dsl_test.go [hunk 3]
```diff
@@ -45,9 +110,36 @@ func TestBuilder(t *testing.T) {
...
+					SetOperationID(ex.Paths[pathWithID].Get.OperationID).
+					AddParameters(
+						ogen.NewParameter().
+							InPath().
+							SetName(ex.Paths[pathWithID].Get.Parameters[0].Name).
+							SetDescription(ex.Paths[pathWithID].Get.Parameters[0].Description).
+							SetRequired(true),
+						ogen.NewParameter().SetRef(authQ.LocalRef()),
+						ogen.NewParameter().SetRef(authH.LocalRef()),
+						ogen.NewParameter().SetRef(csrf.LocalRef()),
```
Comment (a8m) [inline dsl_test.go]: Same here. Consider changing the `Parameter` to an interface (`NamedParam`/`RefParam`/`SimpleParam`), or provide other way that is less verbose to use them here.

```go
AddParameters(
    ogen.NewParamater()...,
    authq.RefParam(),
    // ...
)

// Or using interfaces.
AddParameters(
    ogen.NewParamater()...,
    authq,
    csrf,
    // ...
)
```

File: dsl_test.go (hunk 1 above)
Comment (masseelch) [reply dsl_test.go]: I think this is not necessary. This is only done here like this to ease construction I'd the expected result for the test. The end user will not construct the parameter map like this.

File: dsl_test.go (hunk 2 above)
Comment (masseelch) [reply dsl_test.go]: Thought about that. But I wanted to make sure that the user knows the last call changes the struct type from `Parameter` to `namedParameter`. 

I'll think about it, maybe I find another solution less ambiguous.

**Comment by masseelch** [comment]
@a8m, @ernado

**Pull request merged**